src/memory.js

```diff
@@ -1,18 +1,32 @@
 class MemoryGame {
   constructor(cards) {
     this.cards = cards;
-    // add the rest of the class properties here
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
   }
 
   shuffleCards() {
-    // ... write your code here
+    if (!this.cards){
+      return undefined;
+    } else {
+      return this.cards.sort(() => Math.random() - 0.5);
+    }
   }
 
   checkIfPair(card1, card2) {
-    // ... write your code here
+    this.pairsClicked++;
+
+    if (card1 === card2) {
+      this.pairsGuessed++;
+      return true;
+    } else {
+      return false;
+    }
   }
 
   checkIfFinished() {
-    // ... write your code here
+    const pairsArr = this.cards.length / 2;
+    return this.pairsGuessed === pairsArr;
```

Comment:
Esto esta de 10!

Al ser pocas lineas lo puedes poner de una, pero esta perfect!

```suggestion
    return this.pairsGuessed === this.cards.length / 2;
```